db/builder.cc

```diff
@@ -371,15 +374,19 @@ Status BuildTable(
 
     constexpr IODebugContext* dbg = nullptr;
 
-    Status ignored = fs->DeleteFile(fname, IOOptions(), dbg);
-    ignored.PermitUncheckedError();
+    assert(table_file_created == fs->FileExists(fname, IOOptions(), dbg).ok());
```

Comment:
Is this always guaranteed to be true? What if the file system starts to malfunction just before this `FileExists()` call? Note that file system can be distributed file system and/or NFS, etc.
I think we (try to) use `assert` only to guard against logic errors.

Reply:
> Is this always guaranteed to be true? What if the file system starts to malfunction just before this `FileExists()` call? Note that file system can be distributed file system and/or NFS, etc. I think we (try to) use `assert` only to guard against logic errors.

Ok, I have removed the potentially dangerous assert in the latest commit.